arch/arm/core/aarch32/userspace.S

```diff
@@ -422,9 +455,29 @@ dispatch_syscall:
     lsl r6, #2
     add ip, r6
     ldr ip, [ip]	/* load table address */
+
+#if defined(CONFIG_CPU_CORTEX_R)
+    /*
+     * We can only be in this system call handling code if interrupts were
+     * enabled.  This is because we would only come down this path if we were
+     * actively running in user state, and user state CANNOT disable external
+     * interrupts via irq_lock().  We want external interrupts enabled while
```

Comment:
this is all good, it's undefined behavior to invoke a syscall trap from supervisor mode on any arch, and syscalls themselves need to be preemptible